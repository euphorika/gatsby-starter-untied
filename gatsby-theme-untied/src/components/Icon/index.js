import React from 'react'
import PropTypes from 'prop-types'

const Icon = ({ svg, className, fill, viewBox, xmlns, xmlnsXlink, path }) => (
  <svg
    className={className}
    viewBox={viewBox}
    xmlns={xmlns}
    xmlnsXlink={xmlnsXlink}
  >
    <path d={path} fill={fill} />
  </svg>
)

Icon.defaultProps = {
  xmlns: 'http://www.w3.org/2000/svg',
  xmlnsXlink: 'http://www.w3.org/1999/xlink',
}

Icon.propTypes = {
  svg: PropTypes.node,
  fill: PropTypes.string,
  viewBox: PropTypes.string,
  xmlns: PropTypes.string,
  xmlnsXlink: PropTypes.string,
}

export default Icon
