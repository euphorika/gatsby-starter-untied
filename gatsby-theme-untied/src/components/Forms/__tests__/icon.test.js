import React from "react"
import renderer from "react-test-renderer"
import Icon from "../icon"

describe("Icon", () => {
  it("renders info correctly", () => {
    const tree = renderer.create('<Icon type="info" />')
    expect(tree).toMatchSnapshot()
  })

  it("renders warn correctly", () => {
    const tree = renderer.create('<Icon type="warn" />')
    expect(tree).toMatchSnapshot()
  })

  it("renders error correctly", () => {
    const tree = renderer.create('<Icon type="error" />')
    expect(tree).toMatchSnapshot()
  })
})
