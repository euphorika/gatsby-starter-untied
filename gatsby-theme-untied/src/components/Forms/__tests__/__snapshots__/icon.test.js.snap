// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders error correctly 1`] = `"<Icon type=\\"error\\" />"`;

exports[`Icon renders info correctly 1`] = `"<Icon type=\\"info\\" />"`;

exports[`Icon renders warn correctly 1`] = `"<Icon type=\\"warn\\" />"`;
